osu.Game/Online/SignalRUnionWorkaroundResolver.cs

```diff
@@ -35,7 +35,9 @@ public class SignalRUnionWorkaroundResolver : IFormatterResolver
             typeof(TeamVersusUserState),
         };
 
-        private static readonly IReadOnlyDictionary<Type, IMessagePackFormatter> formatter_map = new Dictionary<Type, IMessagePackFormatter>
+        private static IReadOnlyDictionary<Type, IMessagePackFormatter> formatterMapBacking;
```

Comment:
I am not at all comfortable with dropping the `readonly` spec from this.

Given that both `SignalRUnionWorkaroundResolver.BASE_TYPES` and `SignalRUnionWorkaroundResolver.DERIVED_TYPES` are actually _only ever_ used by `SignalRDerivedTypeWorkaroundJsonConverter`, and _never_ used by `SignalRUnionWorkaroundResolver`, I would much rather just move those constants there.

Reply:
That was my plan initially, but I wasn't sure moving the constants around to fix this would be a comfortable solution really. I thought about making a static class listing the base and derived types would rather be more safe (not sure what to call it though).

Reply:
That would work for me as well. Almost anything's better than a mutable static dictionary.

Reply:
That's actually how it was originally in my work on this. Agree with separating out.